# HG changeset patch
# User Pantxo Diribarne <pantxo.diribarne@gmail.com>
# Date 1424197092 -3600
#      Tue Feb 17 19:18:12 2015 +0100
# Node ID 99ef3e58214ea4e0e640266eec335f319c7d5a57
# Parent  1687269e31e42d86e2cca2a26035ae9d830a0bdd
Run children listeners after any adoption/deletion (bug #44242)

* graphics.in.h (base_properties::adopt, base_properties::remove_child):
                 call children.run_listener() after each individual children
                 adoption/deletion.

diff --git a/libinterp/corefcn/graphics.in.h b/libinterp/corefcn/graphics.in.h
--- a/libinterp/corefcn/graphics.in.h
+++ b/libinterp/corefcn/graphics.in.h
@@ -2525,22 +2525,26 @@ public:
     return false;
   }
 
   bool is_modified (void) const { return is___modified__ (); }
 
   virtual void remove_child (const graphics_handle& h)
   {
     if (children.remove_child (h.value ()))
-      mark_modified ();
+      {
+        children.run_listeners ();
+        mark_modified ();
+      }
   }
 
   virtual void adopt (const graphics_handle& h)
   {
     children.adopt (h.value ());
+    children.run_listeners ();
     mark_modified ();
   }
 
   virtual graphics_toolkit get_toolkit (void) const;
 
   virtual Matrix
   get_boundingbox (bool /*internal*/ = false,
                    const Matrix& /*parent_pix_size*/ = Matrix ()) const
